addons/markers/functions/fnc_onButtonClickConfirm.sqf

```diff
@@ -14,61 +14,98 @@
  *
  * Public: No
  */
+
 params ["_buttonOk"];
 
 private _display = ctrlParent _buttonOk;
 private _description = _display displayCtrl IDC_INSERT_MARKER;
 private _aceTimestamp = _display displayCtrl IDC_ACE_INSERT_MARKER_TIMESTAMP;
 
-// handle timestamp
+// Handle timestamp
 if (cbChecked _aceTimestamp && {ACE_player call FUNC(canTimestamp)}) then {
-    // determine marker timestamp based on time settings
-    private _time = switch (GVAR(timestampTimezone)) do {
+    // Determine marker timestamp based on time settings
+    (switch (GVAR(timestampTimezone)) do {
         case 1: {
-            systemTime params ["", "", "", "_hour", "_min", "_sec"];
-            _hour + _min/60 + _sec/3600
+            systemTime select [3]
         };
         case 2: {
-            systemTimeUTC params ["", "", "", "_hour", "_min", "_sec"];
-            _hourOffset = round (GVAR(timestampUTCOffset));
+            systemTimeUTC params ["", "", "", "_hour", "_min", "_sec", "_msec"];
+
+            private _hourOffset = round GVAR(timestampUTCOffset);
             _hour = _hour + _hourOffset;
 
-            // add or subtract minutes offset based on the negative or positive timezone
-            _min = if (_hourOffset < 0) then { _min - GVAR(timestampUTCMinutesOffset) } else { _min + GVAR(timestampUTCMinutesOffset) };
+            // Add or subtract minutes offset based on the negative or positive timezone
+            if (GVAR(timestampUTCMinutesOffset) != 0) then {
+                _min = if (_hourOffset < 0) then { _min - GVAR(timestampUTCMinutesOffset) } else { _min + GVAR(timestampUTCMinutesOffset) };
 
-            // prevent the timestamp from exceeding 24 hours or going below 0 hours
-            _time = ((_hour + _min/60 + _sec/3600) % 24 + 24) % 24;
-            _time
+                // Add/remove extra hours from minutes
+                _hour = _hour + floor (_min / 60);
+                _min = (_min % 60 + 60) % 60; // ensure that minutes are between 0 and 59 (included)
+            };
+
+            [(_hour % 24 + 24) % 24, _min, _sec, _msec] // ensure that hours are between 0 and 23 (included)
         };
         default {
-            dayTime
+            private _daytime = dayTime;
+
+            private _hour = floor _daytime;
+            private _min = floor ((_daytime - _hour) * 60);
+            private _sec = floor ((((_daytime - _hour) * 60) - _min) * 60);
+            private _msec = floor ((((((_daytime - _hour) * 60) - _min) * 60) - _sec) * 1000);
+
+            [_hour, _min, _sec, _msec]
         };
-    };
+    }) params ["_hour", "_min", "_sec", "_msec"];
```

Comment:
Gonna be honest, preferred `_time = switch...`. Easier to read.

Reply:
Done.